app/src/main/java/org/simple/clinic/bloodsugar/unitselection/BloodSugarUnitSelectionDialog.kt

```diff
@@ -8,29 +8,39 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.RadioButton
 import android.widget.Button
+import android.widget.RadioButton
 import androidx.appcompat.app.AlertDialog
 import androidx.appcompat.app.AppCompatDialogFragment
 import androidx.fragment.app.FragmentManager
 import com.jakewharton.rxbinding3.view.clicks
 import io.reactivex.Observable
 import io.reactivex.rxkotlin.ofType
 import io.reactivex.subjects.PublishSubject
-import kotlinx.android.synthetic.main.dialog_bloodsugar_selectionunit.*
 import org.simple.clinic.R
 import org.simple.clinic.bloodsugar.BloodSugarUnitPreference
+import org.simple.clinic.databinding.DialogBloodsugarSelectionunitBinding
 import org.simple.clinic.di.injector
 import org.simple.clinic.mobius.MobiusDelegate
 import org.simple.clinic.util.unsafeLazy
 import org.simple.clinic.widgets.ScreenDestroyed
-import org.simple.clinic.widgets.UiEvent
 import javax.inject.Inject
 
 class BloodSugarUnitSelectionDialog : AppCompatDialogFragment(), BloodSugarUnitSelectionUiActions {
 
   private var layout: View? = null
 
+  private var binding: DialogBloodsugarSelectionunitBinding? = null
```

Comment:
We also need to clear this in the appropriate lifecycle method because the fragment will leak memory otherwise.

Reply:
I don't see the `binding` variable being cleared anywhere. Can we clear this to avoid a memory leak?

Reply:
Is `onStop()` the right place to do this? If we put the app in the background and then resume it, the binding should be reinitialized, but it won't. Would `onDismiss()` be a better place to do this?

Reply:
Can we do this in `onDetach` ? Dialog would be detached after this and binding would be reinitialized when the dialog is opened again. What do you think?

Reply:
If we do this in `onDetach()`, then we should be initializing the binding in the corresponding `onAttach()` as well.

Reply:
Understood